ksql/src/main/java/org/apache/zeppelin/ksql/BasicKSQLHttpClient.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zeppelin.ksql;
+
+import org.apache.commons.io.IOUtils;
+
+import java.io.BufferedReader;
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class BasicKSQLHttpClient implements Closeable {
+
+  public static final String UTF_8 = "utf-8";
+
+  interface BasicHTTPClientResponse {
+    void onMessage(int status, String message);
+
+    void onError(int status, String message);
+  }
+
+  private final String jsonData;
+  private final Map<String, Object> formData;
+  private final String type;
+  private final Map<String, String> headers;
+  private final URL url;
+  private HttpURLConnection connection;
+  private final int timeout;
+  private boolean connected;
+
+
+  public BasicKSQLHttpClient(String url, String jsonData, Map<String, Object> formData,
+                 String type, Map<String, String> headers, int timeout)
+      throws IOException {
+    this.url = new URL(url);
+    this.jsonData = jsonData;
+    this.formData = formData;
+    this.type = type;
+    this.headers = headers;
+    this.timeout = timeout;
+    this.connected = false;
+  }
+
+  @Override
+  public void close() throws IOException {
+    connected = false;
+    connection.disconnect();
```

Comment:
It is possible that `connection` is null. Add following
```
if (connection != null) {
```

Reply:
done